Lesson 05/lesson05 script2.js

```diff
@@ -0,0 +1,259 @@
+//     if(elem < 0) {
+//     return elem
+// } 
+// } )
+
+// console.log(ArrayWithNegValues.length)
+
+
+
+
+
+// 5.Знайти кількість непарних позитивних елементів.
+//done
+
+// let arr = [16,-37,54,-4,72,-56,47,4, -16,25,-37,46,4,-51,27,-63,4,-54,76,-4,12,-35,4,47]
+
+// const ArrayWithOddPosValues = arr.filter((elem) => {
+//     if(elem > 0 && elem % 2 !== 0){
+//     return true
+
+// } 
+// } )
+
+// console.log(ArrayWithOddPosValues.length)
+
+
+
+
+
+
+
+// 6. Знайти кількість парних позитивних елементів.
+//done
+
+
+// let arr = [16,-37,54,-4,72,-56,47,4, -16,25,-37,46,4,-51,27,-63,4,-54,76,-4,12,-35,4,47]
+
+// const ArrayWithEvenPosValues = arr.filter((elem) => {
+//     if(elem > 0 && elem % 2 === 0){
+//     return true
+// } 
+// } )
+
+// console.log(ArrayWithEvenPosValues.length)
+
+
+
+
+
+
+
+
+
+// 7. Знайти суму парних позитивних елементів.
+// done 
+
+
+// let arr = [16,-37,54,-4,72,-56,47,4, -16,25,-37,46,4,-51,27,-63,4,-54,76,-4,12,-35,4,47]
+
+// const ArrayWithEvenPosValues = arr.filter((elem) => {
+//     if(elem >= 0 && elem % 2 === 0){
+//     return true
+// } 
+// } )
+// console.log(ArrayWithEvenPosValues)
+
+
+// const result = ArrayWithEvenPosValues.reduce((total, elem) => {
+//     return total + elem            
+// }, 0) 
+
+//  console.log(result);
+
+
+
+
+
+
+//  8. Знайти суму непарних позитивних елементів.
+//done
+
+// let arr = [16,-37,54,-4,72,-56,47,4, -16,25,-37,46,4,-51,27,-63,4,-54,76,-4,12,-35,4,47]
+
+// const ArrayWithOddPosValues = arr.filter((elem) => {
+//     if(elem >= 0 && elem % 2 !== 0){
+//     return true
+// } 
+// } )
+// console.log(ArrayWithOddPosValues)
+
+
+// const amount = ArrayWithOddPosValues.reduce((total, elem) => {
+//     return total + elem            
+// }, 0) 
+
+//  console.log(amount);
+
+
+
+
+
+
+
+// 9. Знайти добуток позитивних елементів.
+//done
+
+// let arr = [16,-37,54,-4,72,-56,47,4, -16,25,-37,46,4,-51,27,-63,4,-54,76,-4,12,-35,4,47]
+
+// const ArrayWithOddPosValues = arr.filter((elem) => {
+//     if(elem >= 0){
+//     return true
+// } 
+// } )
+
+// const amount = ArrayWithOddPosValues.reduce((total, elem) => {
+//     return total * elem            
+// }, 1) 
+
+//  console.log(amount);
+
+
+
+
+// Знайти добуток позитивних елементів _v2.
+// done
+
+// let arr = [16,-37,54,-4,72,-56,47,4, -16,25,-37,46,4,-51,27,-63,4,-54,76,-4,12,-35,4,47]
+//     let result = 1
+//    for (i = 0; i < arr.length; i++){
+//     if (arr[i] > 0 )
+//      result *= arr[i]
+//    }
+//    console.log(result);
+
+
+
+
+
+
+
+//10. Знайти найбільший серед елементів масиву, остальні обнулити.
+
+
+// let arr = [16,-37,54,-4,72,-56,47,4, -16,25,-37,46,4,-51,27,-63,4,-54,76,-4,12,-35,4,47]
+
+// const biggestElemOfArray = arr.reduce((total, elem) => {
+//     return Math.max(total, elem)
+// }, -Infinity)
+
+// console.log(biggestElemOfArray);
+
+// const zeroArray = arr.map((elem) => {
+// if (elem !== biggestElemOfArray) {
+//     return elem = 0
```

Comment:
не слід повертати результат присвоєння
```suggestion
//     return 0
```